tools/windows-node-installer/test/e2e/aws_test.go

```diff
@@ -127,6 +132,53 @@ func setupWindowsInstanceWithResources(t *testing.T) {
 	}
 }
 
+// waitForStatusok waits for the instance to be okay.
+func waitForStatusOk(instanceId string) error {
+	for i := 0; i < retryCount; i++ {
+		instanceStatus, err := getInstanceStatus(instanceId)
+		if err != nil {
+			fmt.Errorf("failed to get the status of the instance: %v", err)
+		}
+		if instanceStatus == "ok" {
+			return nil
+		}
+		time.Sleep(retryInterval)
+	}
+	return fmt.Errorf("failed to obtain the ok status")
+}
+
+// getInstanceStatus returns the status of the instance.
+func getInstanceStatus(instanceId string) (string, error) {
+	ec2Svc := awsProvider.EC2
+	input := &ec2.DescribeInstanceStatusInput{
+		DryRun:              nil,
+		Filters:             nil,
+		IncludeAllInstances: nil,
+		InstanceIds: []*string{
+			&instanceId,
+		},
+		MaxResults: nil,
+		NextToken:  nil,
+	}
+	result, err := ec2Svc.DescribeInstanceStatus(input)
+	if err != nil {
+		return "", fmt.Errorf("failed to obtain the Instance status: %v", err)
+	}
+	// currently we are creating only a single instance which is going to be the
+	// zeroth entry in it.
+	if result.InstanceStatuses[0] == nil {
+		return "", fmt.Errorf("failed to obtain Instance status")
+	}
+	instanceStatus := *result.InstanceStatuses[0].InstanceStatus.Status
```

Comment:
`InstanceStatus.Status` are both pointers. You can't dereference them without checking for nil.